app/services/concerns/curator/filestreams/attacher.rb

```diff
@@ -94,22 +97,21 @@ def ingest_content?(io = {})
         # @param io [Tempfile]
         # @returns [ActiveStorage::Blob]
         def import_file_from_fedora(attachment_attributes, io)
-          ActiveStorage::Blob.create!(
-            key: attachment_attributes['key'],
-            filename: attachment_attributes['file_name'],
-            byte_size: attachment_attributes['byte_size'],
-            checksum: checksum_to_base64(attachment_attributes['checksum_md5']),
-            content_type: attachment_attributes['content_type'],
-            service_name: attachment_attributes['service_name'],
-            metadata: attachment_attributes['metadata']
-          ).tap do |blob|
+          ActiveStorage::Blob.find_or_initialize_by(key: attachment_attributes['key']).tap do |blob|
+            blob.filename =  attachment_attributes['file_name']
+            blob.byte_size = attachment_attributes['byte_size']
+            blob.checksum = checksum_to_base64(attachment_attributes['checksum_md5'])
+            blob.content_type = attachment_attributes['content_type']
+            blob.service_name = attachment_attributes['service_name']
+            blob.metadata = attachment_attributes['metadata']
+          end.tap(&:save!).tap do |blob|
```

Comment:
Double `tap` here intentional?

Reply:
Per these methods https://github.com/rails/rails/blob/e3e3a97ada3f2b4c0c8d48f941c2ea05c859cdda/activestorage/app/models/active_storage/blob.rb#L97

I basically mimic'd how these methods chain their taps. Alos this prevents the blob from getting uploaded if the `save!` call fails